platform/core.windows/test/unit/src/org/netbeans/core/windows/ModeManagerTest.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.core.windows;
+
+import java.awt.GraphicsEnvironment;
+import static junit.framework.TestCase.assertNotNull;
+import junit.framework.TestSuite;
+import org.netbeans.junit.NbTestCase;
+import org.openide.modules.ModuleInfo;
+import org.openide.util.Lookup;
+import org.openide.windows.Mode;
+import org.openide.windows.ModeUtility;
+import org.openide.windows.TopComponent;
+import org.openide.windows.WindowManager;
+
+/**
+ * This test exercises the WindowManager.ModeManager interface, 
+ * as implemented by WindowManagerImpl, through the API of ModeUtility.
+ *
+ * @author Mark Phipps
+ */
+public class ModeManagerTest extends NbTestCase {
```

Comment:
OK, this looks like a test that tests something.

Reply:
Changed name to WindowManagerModeTest.